src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -14,12 +14,317 @@
+    {
+        trace::verbose(_X("Considering %s to probe for %s"), probe_path.c_str(), LIBHOSTPOLICY_NAME);
+        if (to_hostpolicy_package_dir(probe_path, version, candidate))
+        {
+            return true;
+        }
+    }
+
+    // Print detailed message about the file not found in the probe paths.
+    trace::error(_X("Could not find required library %s in %d probing paths:"),
+        LIBHOSTPOLICY_NAME, probe_realpaths.size());
+    for (const auto& path : probe_realpaths)
+    {
+        trace::error(_X("  %s"), path.c_str());
+    }
+    return false;
+}
+
+/**
+ * Given FX location, app binary and specified --depsfile, return deps that contains hostpolicy.dll
+ */
+pal::string_t get_deps_file(
+    const pal::string_t& fx_dir,
+    const pal::string_t& app_candidate,
+    const pal::string_t& specified_deps_file,
+    const runtime_config_t& config)
+{
+    if (config.get_portable())
+    {
+        // Portable app's hostpolicy is resolved from FX deps
+        return fx_dir + DIR_SEPARATOR + config.get_fx_name() + _X(".deps.json");
+    }
+    else
+    {
+        // Standalone app's hostpolicy is from specified deps or from app deps.
+        return !specified_deps_file.empty() ? specified_deps_file : get_deps_from_app_binary(app_candidate);
+    }
+}
+
+/**
+ * Given own location, FX location, app binary and specified --depsfile and probe paths
+ *     return location that is expected to contain hostpolicy
+ */
+bool fx_muxer_t::resolve_hostpolicy_dir(host_mode_t mode,
+    const pal::string_t& own_dir,
+    const pal::string_t& fx_dir,
+    const pal::string_t& app_candidate,
+    const pal::string_t& specified_deps_file,
+    const std::vector<pal::string_t>& probe_realpaths,
+    const runtime_config_t& config,
+    pal::string_t* impl_dir)
+{
+    // Obtain deps file for the given configuration.
+    pal::string_t resolved_deps = get_deps_file(fx_dir, app_candidate, specified_deps_file, config);
+
+    // Resolve hostpolicy version out of the deps file.
+    pal::string_t version = resolve_hostpolicy_version_from_deps(resolved_deps);
+    if (trace::is_enabled() && version.empty() && pal::file_exists(resolved_deps))
+    {
+        trace::warning(_X("Dependency manifest %s does not contain an entry for %s"), resolved_deps.c_str(), _STRINGIFY(HOST_POLICY_PKG_NAME));
+    }
+
+    // Check if the given version of the hostpolicy exists in servicing.
+    if (hostpolicy_exists_in_svc(version, impl_dir))
+    {
+        return true;
+    }
+
+    // Get the expected directory that would contain hostpolicy.
+    pal::string_t expected;
+    if (config.get_portable())
+    {
+        if (!pal::directory_exists(fx_dir))
+        {
+            handle_missing_framework_error(config.get_fx_name(), config.get_fx_version(), fx_dir);
+            return false;
+        }
+        
+        expected = fx_dir;
+    }
+    else
+    {
+        // Standalone apps can be activated by muxer or by standalone host or "corehost"
```

Comment:
Yes. Of course, we're shipping soon right? Tried portable, standalone build/run, xunit, published and unpublished with muxer, corehost and standalone where applicable.